salt/states/macdefaults.py

```diff
@@ -4,30 +4,31 @@
 
 """
 
-import logging
-
 import salt.utils.platform
 
-log = logging.getLogger(__name__)
-
 __virtualname__ = "macdefaults"
 
 
 def __virtual__():
     """
-    Only work on Mac OS
+    Only work on macOS
     """
     if salt.utils.platform.is_darwin():
         return __virtualname__
-    return (False, "Only supported on Mac OS")
+    return (False, "Only supported on macOS")
 
 
-def write(name, domain, value, vtype="string", user=None):
+def write(name, domain, value, vtype=None, name_separator=None, user=None):
     """
     Write a default to the system
 
     name
-        The key of the given domain to write to
+        The key of the given domain to write to.
+        It can be a nested key/index separated by `name_separator`.
+
+    name_separator
+        The separator to use when splitting the name into a list of keys.
+        If None, the name will not be split (Default).
```

Comment:
Needs `..versionadded:: 3008`

Reply:
Done in commit d9243837c283d6659df911e8c2c5a44d7ce125ff